src/adv.py

```diff
@@ -49,3 +48,61 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+    
+print("Welcome", Player.name)
+
+def action(choice): 
```

Comment:
Awesome work creating this action function. How could this be improved to be more DRY?